src/figuresClasses.ts

```diff
@@ -1,11 +1,94 @@
-export interface Figure {}
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
+export interface Figure {
+  readonly shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
 
-export class Triangle implements Figure {}
+export class Triangle implements Figure {
+  public shape = Shape.Triangle;
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
```

Comment:
[CODE KNOWLEDGE] - When you declare a constructor parameter with an access modifier TypeScript implicitly creates a property on the class. You don't need to assign them manually.